### DISCUSSION TITLE
iOS - using textbox with a button.

### DISCUSSION THREAD

**Discussion by robertlitvack** [unoplatform/uno#11356] [Q&A]
I have some XAML that in general looks like this:
```
 <Grid Margin="16"
      CornerRadius="12"
      Background="{StaticResource Color08Brush}">
    <Grid.ColumnDefinitions>
        <ColumnDefinition Width="*" />
        <ColumnDefinition Width="Auto" />
    </Grid.ColumnDefinitions>

    <TextBox Grid.Column="0"
             PlaceholderText="Write a message…"
             Margin="12,12,4,12"
             MaxHeight="92"
             TextWrapping="Wrap"
             Text="{Binding [Input], Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"
             Style="{StaticResource ChatTextBoxStyle}" />
    <Button Grid.Column="1"
            Command="{Binding [SendMessage]}"
            Style="{StaticResource EllipseButtonStyle}"
            Background="{StaticResource Color11Brush}"
            Width="44"
            Height="44"
            Margin="0,0,12,0"
            IsTabStop="False">
        <u:PathControl Style="{StaticResource ForwardArrowPathStyle}"
                       Width="8" />
    </Button>
 </Grid>
```
Everything works in Android. Having issues in iOS. When in the TextBox, the iOS keyboard opens as expected. You type some text and hit the Button to send the message. However, this merely dismisses the iOS keyboard and doesn't fire SendMessage

I see for Xamarin.Forms there's `ios:VisualElement.CanBecomeFirstResponder="True"` but not sure what the corresponding logic is with UNO.

Thanks!

**Comment by agneszitte** [comment]
@robertlitvack Thank you a lot for the details.
To better help, is it possible for you to provide a small test solution so it is easier to reproduce your issue, please?
Also, can you tell us if you are testing on a real iOS device or on an iOS emulator? Which one, please?
Which Uno version are you using for this issue?
Thanks in advance for your feedback!

**Comment by robertlitvack** [comment]
Hi @agneszitte-nventive The solution this integrates into is extremely large so that's not really possible. I'm assuming any simple XAML with the Textbox + Button combo in a grid may cause this issue.

I've tried on emulator and real iOS device - fails in both places.

Are you looking for the Uno.UI version?

Thanks

**Reply by jeromelaban** [reply]
Yes, we're interested in the Uno.UI or Uno.WinUI version. We also need a repro to determine if the simple use cases work properly. If you take the XAML you provided and place it in a simple app from the Uno templates, does it show the same behavior? Also, when you click again on the button after the keyboard closed, does it work?

**Comment by robertlitvack** [comment]
Hello,

The project is using Uno.UI 3.1.32. 

After pressing the button with the keyboard open, the keyboard closes (command doesn't fire). If I press the button afterwards (once keyboard is hidden), everything fires properly.

The framework I'm using was built upon Uno and is relatively lengthy.

I was hoping to see if there was something similar to `CanBecomeFirstResponder` from Xamarin.Forms (https://learn.microsoft.com/en-us/xamarin/xamarin-forms/platform/ios/visualelement-first-responder)

**Reply by jeromelaban** [reply]
Thanks. Uno.UI 3.1 is quite old (~3.5 years) and there may have been fixes since then which is why we're asking for a simple repro.

Now, aside from upgrading to a fairly recent build of Uno, you can use the `ios:CanBecomeFirstResponder="true"` in XAML [using conditional XAML](https://platform.uno/docs/articles/platform-specific-xaml.html), if that can help for your issue. It may not work as-is depending on where the property needs to be set, though.

**Reply by jeromelaban** [reply]
_@robertlitvack make sure to answer on the same thread, it's easier to follow._

Just to make everything clear, Xamarin.Forms is not used in any way in Uno Platform, so there's no references to make for that framework.

After looking further, [CanBecomeFirstResponder is a readonly property](https://developer.apple.com/documentation/uikit/uiresponder/1621130-canbecomefirstresponder) that cannot be modified from XAML using Uno, but in all cases, we're not able to reproduce your issue.

**We need a sample repro** for your code above so we can talk about the same issue, we will not otherwise be able to help further. Create a new uno platform app using the Visual Studio templates, make your code compile in that application, test the behavior and let us know if you can reproduce or not. In all cases, please attach the full app sample code here.

**Reply by robertlitvack** [reply]
_@jeromelaban Sure, will stick to threads_

Yes, that's what I understood from my research is that Xamarin.Forms is not actually in play here. From investigating, it was found that Xamarin.Forms had the exact problem I'm encountering and solved it with the new `ios:CanBecomeFirstResponder="true"` which is write-able.

I will work on a sample repo in the next week and see what the behavior is. I was more hoping this issue may have been encountered before. Will get back to you.

**Comment by robertlitvack** [comment]
If I understand correctly, using `xmlns:xamarin="http:/uno.ui/xamarin` doesn't actually use Xamarin.Forms but rather ensures the code is run for the corresponding entries in the platform support table, correct?

From my research, the `ios:VisualElement.CanBecomeFirstResponder="True"` also requires `xmlns:ios="clr-namespace:Xamarin.Forms.PlatformConfiguration.iOSSpecific;assembly=Xamarin.Forms.Core"`. I'm not sure how to apply that with UNO Framework.

Thanks!

**Comment by robertlitvack** [comment]
Hello,

You can close this issue, as I'm ultimately using a custom framework so reproducing or not won't help. 

Thanks for your time!

**Reply by jeromelaban** [reply]
Thanks for the feedback. If you don't mind my asking, what do you mean by custom framework?

**Reply by robertlitvack** [reply]
There was a custom framework built by a third party. It was before my time so I don't know all the details

Thanks again!

**Comment by robertlitvack** [comment]
@jeromelaban @agneszitte-nventive 
Sorry, I wanted to circle back to this as I've gotten some additional details. The framework here was written by nventive, it is a combination of Uno + Umbrella.

My question: as this is a custom framework (pre-UNO open source), would reproducing this on the current open-source version of UNO even be useful?

**Reply by robertlitvack** [reply]
Hello,

Just wanted to follow-up on this one. Thanks!

**Reply by sasakrsmanovic** [reply]
Hi @robertlitvack The current implementation of Uno Extensions has been inspired by former work that was done on 'umbrella' quite some years ago. Maybe the best way to proceed and have a more comprehensive idea of the task you have at hand would be to have a short call. Happy to sign an NDA if needed - please reach out to me at sasha@platform.uno and I will coordinate with our team.
